### PR TITLE
Fix "Child already has a owner" Assert

### PR DESCRIPTION
**Pull request by kennethpu**
Upgrading to the latest version of Yoga introduced an assert in `YGNodeInsertChild()` that was previously removed accidentally ([see relevant commit](https://github.com/facebook/yoga/commit/2e234473cadb6884a06615c1a5cca82026c47012)). This assert will throw an error if the node we are trying to add already has an owner. Unfortunately, we can fall into this case quite easily if we move a view between different superviews (as demonstrated in the following contrived example). 

```
let viewA = UIView()
let viewB = UIView()
let viewC = UIView()

flex.addItem(viewA)
flex.addItem(viewB)

// Add viewC as a subview of viewA
viewA.flex.addItem(viewC)
flex.layout()

// Add viewC as a subview of viewB instead
viewB.flex.addItem(viewC)
flex.layout() // at this point we will hit the assert
```

~To avoid triggering this assert we should check in `Flex.addItem(_ view: UIView)` whether `view` already has a superview, and if so remove the view's associated yoga node as a child from the superview's associated yoga node. This approach mirror's how UIKit handles adding subviews, and should help ensure that Yoga and UIKit's representation of the view hierarchy remains in sync~

UPDATE - Restoring the changes from this old commit (https://github.com/layoutBox/FlexLayout/commit/c303faae7b7b441872f8d49dbd1424db6ec7953f) should ensure that any lingering parent references are cleaned up before we call `YGNodeInsertChild()`

### CODE REVIEW

File: Sources/Swift/FlexLayout.swift [hunk 1]
```diff
@@ -69,15 +72,23 @@ public final class Flex {
...
+    
+     - Parameter view: The view to be added.
+     - Returns: The flex interface corresponding to the added view.
      */
     @discardableResult
     public func addItem(_ view: UIView) -> Flex {
         if let host = self.view {
+            if let superview = view.superview, superview != host && superview.isYogaEnabled {
+                YGNodeRemoveChild(superview.yoga.node, view.yoga.node)
+            }
```
Comment (kennethpu) [inline Sources/Swift/FlexLayout.swift:91]: Fix is here - check if `view` currently has a superview that isn't the host view, and if so remove it's associated yoga node as a child from the superview's associated yoga node.

**Comment by kennethpu** [comment]
Alternatively, I found this old commit (https://github.com/layoutBox/FlexLayout/commit/c303faae7b7b441872f8d49dbd1424db6ec7953f) where we modified `YGAttachNodesFromViewHierachy()` to remove any lingering parent references - restoring this commit should be a viable fix as well.

**Comment by lucdion** [comment]
Hi @kennethpu,  could you update your branch, I have fixed the CI. Then after, I will be able to merge your PR. Thanks

**Review by lucdion — APPROVED** [review]
Thanks for this important fix!

**Comment by lucdion** [comment]
Your fix has been [released](https://github.com/layoutBox/FlexLayout/releases/tag/2.0.05). Thanks for this change

**Pull request merged**